rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -261,7 +264,8 @@ rmw_destroy_client(rmw_node_t * node, rmw_client_t * client)
     }
     delete info;
   }
-  delete (client);
+  rmw_free(const_cast<char *>(client->service_name));
```

Comment:
https://github.com/ros2/rmw/blob/master/rmw/src/allocators.c#L37

Since the `rmw_free` function just calls free, I think it's ok to remove the null check around the one above.